src/Parallel/Info.hpp

```diff
@@ -7,98 +7,101 @@
 
 #pragma once
 
+#include <cstddef>
+
 /// Functionality for parallelization.
 ///
 /// The functions in namespace `Parallel` that return information on
 /// nodes and cores are templated on DistribObject.  Actions should
 /// use these functions rather than the raw charm++ versions (in the
-/// sys namespace in Utilities/System/ParalleInfo.hpp) so that the
+/// sys namespace in Utilities/System/ParallelInfo.hpp) so that the
 /// mocking framework will see the mocked cores and nodes.
 namespace Parallel {
 
 /*!
  * \ingroup ParallelGroup
  * \brief Number of processing elements.
  */
-template <typename DistribObject>
-int number_of_procs(const DistribObject& distributed_object) {
-  return distributed_object.number_of_procs();
+template <typename T, typename DistribObject>
+T number_of_procs(const DistribObject& distributed_object) {
+  return static_cast<T>(distributed_object.number_of_procs());
```

Comment:
[optional] could add a `static_assert(std::is_integer_v<T>);` to catch casting to weird types. Up to you!

Reply:
I think I'll leave it as is for now. If it becomes an issue in the future I can just change it then.

Reply:
Sounds good!